src/Compilers/Shared/BuildClient.cs

```diff
@@ -69,25 +114,28 @@ internal RunCompilationResult RunCompilation(IEnumerable<string> originalArgumen
             List<string> parsedArgs;
             bool hasShared;
             string keepAliveOpt;
-            string sessionKeyOpt;
             string errorMessageOpt;
-            if (!CommandLineParser.TryParseClientArgs(
+            if (CommandLineParser.TryParseClientArgs(
                     args,
                     out parsedArgs,
                     out hasShared,
                     out keepAliveOpt,
-                    out sessionKeyOpt,
+                    out string commandLinePipeName,
                     out errorMessageOpt))
+            {
+                pipeName = pipeName ?? commandLinePipeName;
```

Comment:
>pipeName ?? commandLinePipeName [](start = 27, length = 31)

Should we prefer `commandLinePipeName` if set?

Reply:
Also a `??=` opportunity?

---
In reply to: [319604674](https://github.com/dotnet/roslyn/pull/38411#discussion_r319604674) [](ancestors = 319604674)

Reply:
No the explicit pipe name passed into the function is intended to override the command line one. 

Note: in reality this doesn't ever happen. The production binaries always pass `null` for the `pipeName`. The override is only used in test code. 

---
In reply to: [319604674](https://github.com/dotnet/roslyn/pull/38411#discussion_r319604674) [](ancestors = 319604674)